ambry-store/src/test/java/com.github.ambry.store/LogSegmentTest.java

```diff
@@ -0,0 +1,702 @@
+public class LogSegmentTest {
+  private static final int STANDARD_SEGMENT_SIZE = 1024;
+
+  private final String tempDirName;
+  private final File tempFile;
+  private final StoreMetrics metrics;
+
+  /**
+   * Sets up a temporary file and directory that can be used.
+   * @throws IOException
+   */
+  public LogSegmentTest()
+      throws IOException {
+    tempFile = File.createTempFile("ambry", ".tmp");
+    tempFile.deleteOnExit();
+    tempDirName = tempFile.getParent();
+    metrics = new StoreMetrics(tempDirName, new MetricRegistry());
+  }
+
+  /**
+   * Deletes the temporary file that was created.
+   */
+  @After
+  public void cleanup() {
+    assertTrue("The temporary file [" + tempFile.getAbsolutePath() + "] could not be deleted", tempFile.delete());
+  }
+
+  /**
+   * Tests appending and reading to make sure data is written and the data read is consistent with the data written.
+   * @throws IOException
+   */
+  @Test
+  public void basicWriteAndReadTest()
+      throws IOException {
+    String segmentName = "log_current";
+    LogSegment segment = getSegment(segmentName, STANDARD_SEGMENT_SIZE);
+    try {
+      assertEquals("Name of segment is inconsistent with what was provided", segmentName, segment.getName());
+      int writeSize = 100;
+      byte[] buf = TestUtils.getRandomBytes(3 * writeSize);
+      // append with buffer
+      int written = segment.appendFrom(ByteBuffer.wrap(buf, 0, writeSize));
+      assertEquals("Size written did not match size of buffer provided", writeSize, written);
+      assertEquals("End offset is not equal to the cumulative bytes written", writeSize, segment.getEndOffset());
+
+      // append with channel
+      segment.appendFrom(Channels.newChannel(new ByteBufferInputStream(ByteBuffer.wrap(buf, writeSize, writeSize))),
+          writeSize);
+      assertEquals("End offset is not equal to the cumulative bytes written", 2 * writeSize, segment.getEndOffset());
+
+      // use writeFrom
+      segment.writeFrom(Channels.newChannel(new ByteBufferInputStream(ByteBuffer.wrap(buf, 2 * writeSize, writeSize))),
+          segment.getEndOffset(), writeSize);
+      assertEquals("End offset is not equal to the cumulative bytes written", 3 * writeSize, segment.getEndOffset());
+
+      readAndEnsureMatch(segment, 0, buf);
+      // test that file size and end offset
+      assertEquals("End offset is not equal to the cumulative bytes written", 3 * writeSize, segment.getEndOffset());
+      assertEquals("Size in bytes is not equal to size of the file", STANDARD_SEGMENT_SIZE, segment.sizeInBytes());
+
+      // ensure flush doesn't throw any errors.
+      segment.flush();
+      // close and reopen segment and ensure persistence.
+      segment.close();
+      segment = new LogSegment(segmentName, new File(tempDirName, segmentName), STANDARD_SEGMENT_SIZE, metrics);
+      segment.state = LogSegment.State.ACTIVE;
+      segment.setEndOffset(buf.length);
+      readAndEnsureMatch(segment, 0, buf);
```

Comment:
This one does not require multiple calls. It is only checking if the data is still around.